File: webflux-dialogue/src/main/java/com/study/webflux/rag/infrastructure/adapter/tts/loadbalancer/TtsEndpointFailureEvent.java
package com.study.webflux.rag.infrastructure.adapter.tts.loadbalancer;

import java.time.Instant;

/**
 * TTS 엔드포인트 영구 장애 이벤트
 */
public class TtsEndpointFailureEvent {
	private final String endpointId;
	private final String errorType;
	private final String errorMessage;
	private final Instant occurredAt;

	public TtsEndpointFailureEvent(String endpointId, String errorType, String errorMessage) {
		this.endpointId = endpointId;
		this.errorType = errorType;
		this.errorMessage = errorMessage;
		this.occurredAt = Instant.now();
	}

	public String getEndpointId() {
		return endpointId;
	}

	public String getErrorType() {
		return errorType;
	}

	public String getErrorMessage() {
		return errorMessage;
	}

	public Instant getOccurredAt() {
		return occurredAt;
	}

	@Override
	public String toString() {
		return String.format("TtsEndpointFailureEvent{endpointId='%s', errorType='%s', errorMessage='%s', occurredAt=%s}",
			endpointId, errorType, errorMessage, occurredAt);
	}
}

File: infrastructure/src/main/kotlin/com/miyou/app/infrastructure/dialogue/adapter/tts/loadbalancer/TtsEndpointFailureEvent.kt
package com.miyou.app.infrastructure.dialogue.adapter.tts.loadbalancer

import java.time.Instant

data class TtsEndpointFailureEvent(
    val endpointId: String,
    val errorType: String,
    val errorMessage: String,
    val occurredAt: Instant = Instant.now(),
) {
    init {
        require(endpointId.isNotBlank()) { "endpointId must not be blank" }
        require(errorType.isNotBlank()) { "errorType must not be blank" }
        require(errorMessage.isNotBlank()) { "errorMessage must not be blank" }
    }
}
